linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_profile/ansible/shared.yml

```diff
@@ -5,24 +5,37 @@
 # disruption = low
 {{{ ansible_instantiate_variables("var_accounts_user_umask") }}}
 
-- name: Check if umask is already set
+- name: '{{{ rule_title }}} - Locate profile configuration files where umask is defined'
+  ansible.builtin.find:
+    paths:
+      - /etc/profile.d
+    patterns:
+      - sh.local
+      - '*.sh'
+    contains: ^[\s]*umask\s+\d+
+  register: result_profile_d_files
+
+- name: '{{{ rule_title }}} - Replace existing umask value in files from /etc/profile.d'
+  ansible.builtin.replace:
+    path: '{{ item.path }}'
+    regexp: ^(\s*)umask\s+\d+
+    replace: \1umask {{ var_accounts_user_umask }}
+  loop: '{{ result_profile_d_files.files }}'
+  register: result_umask_replaced_profile_d
+  when: result_profile_d_files.matched
+
+- name: '{{{ rule_title }}} - Ensure umask is set in /etc/profile if not already set elsewhere'
   ansible.builtin.lineinfile:
+    create: true
+    mode: 0644
     path: /etc/profile
-    regexp: (^[\s]*umask)\s+(\d+)
-    state: absent
-  check_mode: yes
-  changed_when: false
-  register: result_umask_is_set
+    line: umask {{ var_accounts_user_umask }}
+  when:
+    - not result_profile_d_files.matched
 
-- name: Replace user umask in /etc/profile
+- name: '{{{ rule_title }}} - Ensure umask value for all existing umask definition in /etc/profile'
```

Comment:
Per the [style guide](https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#ansible) this should be in title case.

Reply:
fixed